src/lib/storage/lz4/lz4_encoder.hpp

```diff
@@ -49,30 +53,35 @@ class LZ4Encoder : public SegmentEncoder<LZ4Encoder> {
     });
 
     /**
-     * Use the LZ4 high compression API to compress the copied values. As C-library LZ4 needs raw pointers as input
-     * and output. To avoid directly handling raw pointers we use std::vectors as input and output. The input vector
-     * contains the data that needs to be compressed and the output vector is allocated enough memory to contain
-     * the compression result. Via the .data() call we can supply LZ4 with raw pointers to the memory the vectors use.
-     * These are cast to char-pointers since LZ4 expects char pointers.
+     * Pre-compute a zstd dictionary if the input data is split among multiple blocks. This dictionary allows
+     * independent compression of the blocks, while maintaining a good compression ratio.
```

Comment:
why is that good?

Reply:
I'm not sure whether you've seen my comment from a few days ago where I tried to explain why we train a dictionary: https://github.com/hyrise/hyrise/pull/1521#issuecomment-470761144
If that leaves you with questions, what exactly are they?

Reply:
This should be included in the comment. Once it#s merged, anything in the pr comments will never be looked again.